lua/bufferline/buffer.lua

```diff
@@ -30,6 +30,10 @@ local utils = require'bufferline.utils'
 
 --- @alias bufferline.buffer.activity.name 'Inactive'|'Alternate'|'Visible'|'Current'
 
+--- A bidirectional map of activities to activity names
+--- @type {[bufferline.buffer.activity]: bufferline.buffer.activity.name, [bufferline.buffer.activity.name]: bufferline.buffer.activity}
+local activities = vim.tbl_add_reverse_lookup {'Inactive', 'Alternate', 'Visible', 'Current'}
```

Comment:
No more `if activity == 3 then hl = 'BufferVisibleFoo' end`. Just do `'BufferVisible' .. activities[activity] .. 'Foo'`.

A few snippets of code were simplified by this, and one was actually _fixed_ because there was an oversight in implementing 4749af7dcfff4c4e8305bac4bd3734dc256d4d91. No chance of any oversights any more.